NetworkPkg/HttpBootDxe/HttpBootImpl.c

```diff
@@ -400,13 +401,37 @@ HttpBootLoadFile (
   //
   // Load the boot file into Buffer
   //
-  Status = HttpBootGetBootFile (
-             Private,
-             FALSE,
-             BufferSize,
-             Buffer,
-             ImageType
-             );
+  for (Retries = 0; Retries < 30; Retries++) {

```

Comment:
Define the max number of retries as a macro or PCD. Is there a reason you have used 30?

Reply:
The value of 30 was just a guess. During my validations, I never saw the download to be retried more than 2 times. I will change as requested.

Reply:
Lets keep it at a lower level then. Maybe 5? Keeping it as a PCD would help with letting a developer modify it if its needed to be higher.

Reply:
Is there any guidance from HTTP specs on number of retries and time to wait between retries.

Reply:
Good point. Will check if the specs mention a recommendation and/or how other sw implements the retry limit.

Reply:
Number of retries - No specific value requires as a part of the spec. In general, it is common to see a default maximum number of retries set to 3 or 5. It can vary dependent on client/server config. The updated code uses 5 so we should be good.

Time to wait between retries - Same here, no default value defined in spec. Commonly see 1-2 seconds. @leandrobecker-pst  can we make this a PCD as well with a default of 2 seconds instead of the currently implemented 3?

Reply:
> Number of retries - No specific value requires as a part of the spec. In general, it is common to see a default maximum number of retries set to 3 or 5. It can vary dependent on client/server config. The updated code uses 5 so we should be good.
> 
> Time to wait between retries - Same here, no default value defined in spec. Commonly see 1-2 seconds. @leandrobecker-pst can we make this a PCD as well with a default of 2 seconds instead of the currently implemented 3?

Sure, no problem!